platform/darwin/src/MGLLight.mm

```diff
@@ -0,0 +1,126 @@
+{
+    return nil;
+}
+
+- (instancetype)initWithMBGLLight:(const mbgl::style::Light &)mbglLight
+{
+    if (self = [super init]) {
+        auto propertyValue = mbglLight.getAnchor();
+        MGLStyleValue<NSValue *> *lightAnchorType;
+        if (propertyValue.isUndefined()) {
+            mbgl::style::PropertyValue<mbgl::style::LightAnchorType> defaultAnchor = mbglLight.getDefaultAnchor();
+            lightAnchorType = MGLStyleValueTransformer<mbgl::style::LightAnchorType, NSValue *, mbgl::style::LightAnchorType, MGLLightAnchor>().toEnumStyleValue(defaultAnchor);
+        } else {
+            lightAnchorType = MGLStyleValueTransformer<mbgl::style::LightAnchorType, NSValue *, mbgl::style::LightAnchorType, MGLLightAnchor>().toEnumStyleValue(propertyValue);
+        }
+        
+        if ([lightAnchorType isKindOfClass:[MGLConstantStyleValue class]]) {
+            NSValue *anchorValue = ((MGLConstantStyleValue *)lightAnchorType).rawValue;
+            _lightAnchor = [anchorValue MGLLightAnchorValue];
+        }
+        
+        _lightAnchorTransition = [self transitionFromOptions:mbglLight.getAnchorTransition()];
+        
+        auto positionValue = mbglLight.getPosition();
+        if (positionValue.isUndefined()) {
+            _position = MGLStyleValueTransformer<mbgl::style::Position, NSValue *>().toStyleValue(mbglLight.getDefaultPosition());
+        } else {
+            _position = MGLStyleValueTransformer<mbgl::style::Position, NSValue *>().toStyleValue(positionValue);
+        }
+        
+        _positionTransiton = [self transitionFromOptions:mbglLight.getPositionTransition()];
+        
+        auto colorValue = mbglLight.getColor();
+        if (colorValue.isUndefined()) {
+            _color = MGLStyleValueTransformer<mbgl::Color, MGLColor *>().toStyleValue(mbglLight.getDefaultColor());
+        } else {
+            _color = MGLStyleValueTransformer<mbgl::Color, MGLColor *>().toStyleValue(colorValue);
+        }
+        
+        _colorTransiton = [self transitionFromOptions:mbglLight.getColorTransition()];
+        
+        auto intensityValue = mbglLight.getIntensity();
+        if (intensityValue.isUndefined()) {
+            _intensity = MGLStyleValueTransformer<float, NSNumber *>().toStyleValue(mbglLight.getDefaultIntensity());
+        } else {
+            _intensity = MGLStyleValueTransformer<float, NSNumber *>().toStyleValue(intensityValue);
+        }
+        
+        _intensityTransition = [self transitionFromOptions:mbglLight.getIntensityTransition()];
+    }
+
+    return self;
+
+}
+
+- (MGLTransition)transitionFromOptions:(const mbgl::style::TransitionOptions &)options {
+    MGLTransition transition;
+    transition.duration = MGLTimeIntervalFromDuration(options.duration.value_or(mbgl::Duration::zero()));
+    transition.delay = MGLTimeIntervalFromDuration(options.delay.value_or(mbgl::Duration::zero()));
+    
+    return transition;
+}
+
+- (mbgl::style::Light)toMBGLLight
```

Comment:
Method names don't begin with prepositions in Objective-C; if the method returns a value without side effects, the name should generally be a noun phrase. `mbgLight` would be a fine name for this method.